### ISSUE TITLE
List of potential CPU architectures

### ISSUE THREAD

**Issue by luser** [saagarjha/dummy_thicc#5]
Just for the record, the full list of `cpu_type_t` and `cpu_subtype_t` values can be found in `mach/machine.h`:
https://opensource.apple.com/source/xnu/xnu-4570.41.2/osfmk/mach/machine.h.auto.html

This means presumably this project is potentially missing out on:

- m68k (seems to be in #3)
- HPPA (PA-RISC, I guess [NeXT supported it at some point](https://www.openpa.net/nextstep_pa-risc.html))
- sparc [same deal](https://wiki.preterhuman.net/NeXTSTEP_for_SPARC_FAQ)
- I860 [This is absolutely wild](https://en.wikipedia.org/wiki/NeXTdimension)
- MC98000 / MC88000 (I guess these are PowerPC?)

**Comment by saagarjha** [comment]
Yep, it’s also part of the macOS SDK and available via `man 3 arch` :) Once #3 is merged in we should have at least one m68k; I (and others) have looked into PA-RISC at some point but I don’t think anyone spent enough time in QEMU yet trying to get that to work and I know literally nobody who has a physical machine for that, although I’m hopeful somebody might stop by someday. Same story for SPARC. I didn’t even know NeXTdimension ran Mach! I’ll have to take a look sometime if someone has put in emulation effort here. 

I should note that some of the architectures and certainly many of the subarchitectures in the file will never be included because no official physical hardware existed for them, or the Mach kernel or toolchains never bothered to tell the difference (for examplem VAX, the many i386 subtypes, unless you have something totally wild to show me). The “easy” ones we still have left  are armv6 (iPhone, iPhone 3G), armv7 (iPhone 3GS/4/4S, Apple Watch Series 1/2/3 and the original one), ppc740 (G3, probably coming in #1) and ppc970 (G5). And then we have the ones that NeXTSTEP ran on that we’ve mentioned. I think we’re we may want to look after that is the various Rhapsody/Mac OS X Server releases and see if the can cover the rest of the PowerPC spread. 

One thing I’m not sure yet is the new “PAC ABI” (and as of A14, “PAK” whatever that is. I thing for the purposes of the Mach loader they are may be different subtypes in places that matter but I’m not completely sure. And I think shoving 64+64 arm64e slices in the binary, all with different PAC ABIs, would make the project much less interesting anyways.

**Comment by NattyNarwhal** [comment]
88k was Motorola's short-lived 68k successor RISC. It didn't do so hot and they switched to making PPC processors. There was a prototype (unsure how far it got) of a NeXT RISC Workstation using the 88k. (I suspect 98k is a variant of it?)

**Comment by ohmantics** [comment]
Pretty much all the NeXTSTEP ones can be generated using cctools's as and ld.
